community_tasks/aimo_evals.py

```diff
@@ -0,0 +1,72 @@
+# MIT License
+
+# Copyright (c) 2024 The HuggingFace Team
+# Copyright (c) 2024 Philip May, Deutsche Telekom AG
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# ruff: noqa: F405, F403, F401
+"""
+Custom evaluation tasks for lighteval.
+
+This file generally create just a TASKS_TABLE and TASKS_GROUPS which are then imported by LightEval.
+This module implements the ...
+"""
+
+from lighteval.tasks.lighteval_task import LightevalTaskConfig
+from lighteval.tasks.requests import Doc
+
+
+task = LightevalTaskConfig(
+    name="aimo_progress_prize_1",
+    prompt_function="prompt",
+    suite=["community"],
+    hf_subset="",
+    hf_repo="lighteval/aimo_progress_prize_1",
+    hf_avail_splits=["train"],
+    evaluation_splits=["train"],
+    few_shots_split="train",
+    few_shots_select="sequential",
+    metric=["perfect_exact_match"],
+    generation_size=5,
```

Comment:
This is a generative task so we need a lot of tokens :)
```suggestion
    generation_size=2048,
```

Reply:
Interesting, I thought the answers were systematically numerical and a couple of tokens - do you want to allow/encourage cot setups for example?

Reply:
Should we, more generally, allow the generation size to be changed if we expect it to be launched with CoT?

Reply:
The answers are indeed numerical like `\\boxed{100}` but they often appear at the end of a long generative process (possibly involving CoT and/or code execution). You can see some examples here: https://huggingface.co/datasets/HuggingFaceH4/details-AI-MO_deepseek-math-7b-sft/viewer/aimo_v03.00.mini_math.2024-04-26T00-57-26